module-apps/application-messages/windows/SMSTemplatesWindow.cpp

```diff
@@ -112,12 +112,24 @@ namespace gui
         }
 
         if (auto switchData = dynamic_cast<SMSTemplateRequest *>(data)) {
+            ignoreWindowsOfThisAppOnSwitchBack = switchData->ignoreCurrentWindowOnStack;
             smsTemplateRequestHandler(switchData);
         }
 
         if (auto switchData = dynamic_cast<SMSSendTemplateRequest *>(data); switchData != nullptr) {
+            ignoreWindowsOfThisAppOnSwitchBack = switchData->ignoreCurrentWindowOnStack;
             smsSendTemplateRequestHandler(switchData);
         }
     }
 
+    bool SMSTemplatesWindow::onInput(const InputEvent &inputEvent)
+    {
+        if (inputEvent.isShortRelease(KeyCode::KEY_RF) && ignoreWindowsOfThisAppOnSwitchBack) {
+            auto app = dynamic_cast<app::ApplicationMessages *>(application);
+            assert(app != nullptr);
+            app::manager::Controller::switchBack(
+                app, std::make_unique<app::manager::SwitchBackRequest>("ApplicationCall", nullptr, true));
```

Comment:
I'm not a fan of adding hard dependencies like calling back to "ApplicationCall". Maybe it will be possible to intercept the sender in the "SMSSendTemplateRequest" message?

Reply:
You right, I changed this